### PR TITLE
feat: Add basic playlist with extra information display for the song

### PR DESCRIPTION
**Pull request by santiagomonsech**
Addressing [#10 ](https://github.com/emeks-studio/elune-pwa/issues/2) (Add Basic Playlist functionality)

    Add Playlist component to host player ui
    Add mocked list of songs
    Add information for the current song being played
    Add stylesheets for the components
    Add Stop control

TODO:

   - Code Review
   - Analyze if refactor is nedded

### CODE REVIEW

File: src/modules/player/AudioPlayer/AudioPlayer.res [hunk 1]
```diff
@@ -32,64 +52,94 @@ module AudioControls = {
...
+          <progress value={pctPlayed} max="100" />
+          <div className="progressTimer">{React.string(duration)}</div>
+        </div>
+      </div>
+      <div className="Playercontrols">
+        <button onClick=stopAudio> {React.string("Stop")} </button>
+        {isPlaying
+          ? <button onClick=pauseAudio> {React.string("Pause")} </button>
+          : <>
+                <button onClick=playAudio> {React.string("Play")} </button>
```
Comment (randomeduc) [inline src/modules/player/AudioPlayer/AudioPlayer.res:103]: Faltaron las llaves `{ }` 😅

File: src/modules/player/AudioPlayer/PlayList.res [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+@scope("JSON") @val external parseIntoMyData: string => songs = "parse"
+
+module PlayListSource = {
+    let getSongs = () => {
+        let playerSongs = parseIntoMyData(Songs.songs)
+        playerSongs.songs
+    };
+};
+
+@react.component
```
Comment (randomeduc) [inline src/modules/player/AudioPlayer/PlayList.res:22]: Por qué esta aislado del _scope_ del `module PlaytListSource`?

File: src/modules/player/AudioPlayer/AudioPlayer.res [hunk 3]
```diff
@@ -1,95 +1,191 @@
+%%raw("import './AudioPlayer.css'")
+
+@scope("JSON") @val external parseIntoMyData: string => Playlist.songs = "parse"
```
Comment (mirko-plowtech) [inline src/modules/player/AudioPlayer/AudioPlayer.res:3]: mmm creo que esto se puede mejorar usando alguna lib, despues paso algun ejemplo

File: src/modules/player/AudioPlayer/AudioPlayer.res [hunk 4]
```diff
@@ -1,95 +1,191 @@
...
 
+let formatTime = (secs: int): string => {
+  let minutes: int = secs / 60
+  let seconds: int = mod(secs, 60) / 1
+
+  let formattedMinutes = Belt.Int.toString(minutes)
+  let formattedSeconds = Belt.Int.toString(seconds)
+
+  if formattedSeconds->String.length < 2 {
+    formattedMinutes ++ ":0" ++ formattedSeconds
```
Comment (mirko-plowtech) [inline src/modules/player/AudioPlayer/AudioPlayer.res:29]: [primitive-types#string-interpolation](https://rescript-lang.org/docs/manual/latest/primitive-types#string-interpolation)

```reason
`${formattedMinutes}:0${formattedSeconds}`
```

File: src/modules/player/AudioPlayer/AudioPlayer.res [hunk 5]
```diff
@@ -1,95 +1,191 @@
...
+    </div>
+  }
+}
+
 module AudioControls = {
   @react.component
-  let make = (~audioRef: React.ref<Js.Nullable.t<Dom.element>>) => {
+  let make = (~audioRef: React.ref<Js.Nullable.t<Dom.element>>, ~song: Playlist.song) => {
     let (isPlaying, setIsPlaying) = React.useState(_ => false)
     let (pctPlayed, setPctPlayed) = React.useState(_ => "0")
```
Comment (mirko-plowtech) [inline src/modules/player/AudioPlayer/AudioPlayer.res:59]: `pct` stands for?

File: src/modules/player/AudioPlayer/AudioPlayer.res [hunk 6]
```diff
@@ -1,95 +1,191 @@
...
+      }}
+    </div>
+  }
+}
+
 module AudioControls = {
   @react.component
-  let make = (~audioRef: React.ref<Js.Nullable.t<Dom.element>>) => {
+  let make = (~audioRef: React.ref<Js.Nullable.t<Dom.element>>, ~song: Playlist.song) => {
     let (isPlaying, setIsPlaying) = React.useState(_ => false)
```
Comment (mirko-plowtech) [inline src/modules/player/AudioPlayer/AudioPlayer.res:58]: maybe all these different state variables could be grouped in a more meaningful (single) type with different states

File: src/modules/player/AudioPlayer/AudioPlayer.res (hunk 5 above)
Comment (bluesd) [reply src/modules/player/AudioPlayer/AudioPlayer.res:59]: percentage, it was present since the previous PR

**Pull request merged**